src/java/org/apache/cassandra/locator/TokenMetadata.java

```diff
@@ -253,34 +266,39 @@ public void updateHostId(UUID hostId, InetAddress endpoint)
     public UUID getHostId(InetAddress endpoint)
     {
         lock.readLock().lock();
+        publicLock.readLock().lock();
         try
         {
             return endpointToHostIdMap.get(endpoint);
         }
         finally
         {
             lock.readLock().unlock();
+            publicLock.readLock().unlock();
```

Comment:
Should these be unlocked in reverse order of locking?

```suggestion
            publicLock.readLock().unlock();

            lock.readLock().unlock();
```